spark/v4.0/spark/src/test/java/org/apache/iceberg/spark/actions/TestRewriteTablePathsAction.java

```diff
@@ -1209,6 +1213,191 @@ public void testNestedDirectoryStructurePreservation() throws Exception {
     assertThat(targetPath2).startsWith(targetTableLocation());
   }
 
+  @Test
+  public void testFullRewriteUpdatesAllManifestLengthsInManifestList(
+      @TempDir Path rootTargetLocation) throws Exception {
+    String location = newTableLocation();
+    Table sourceTable = createTableWithSnapshots(location, 10);
+
+    addAnyPositionDelete(
+        sourceTable, removePrefix(sourceTable.location() + "/data/deeply/nested/deletes.parquet"));
+
+    Map<String, Long> manifestSizesBeforeRewrite =
+        sourceTable.currentSnapshot().allManifests(sourceTable.io()).stream()
+            .collect(Collectors.toMap(m -> fileName(m.path()), m -> m.length()));
+
+    // Rewrite table metadata to a location that's much longer than the original in order
+    // to make manifests larger
+    String targetLocation = toAbsolute(rootTargetLocation) + generateLongNestedPath(25);
```

Comment:
I think you can actually just move this into the function above and no need to keep `@TempDir Path rootTargetLocation` for each tests

```java
  protected String targetTableLocation() {
    return toAbsolute(targetTableDir) + generateLongNestedPath(5);
  }
```

Reply:
I tried with longer targetTableLocation for all tests and they seem to be happy 😉  But I am fine with `longTargetTableLocation`